x-pack/plugins/reporting/server/deprecations/migrate_existing_indices_ilm_policy.ts

```diff
@@ -27,7 +27,7 @@ export const getDeprecationsInfo = async (
     elasticsearchClient: esClient.asInternalUser,
   });
 
-  if (migrationStatus !== 'ok') {
+  if (migrationStatus !== 'ok' || reportingCore.getConfig().statefulSettings.enabled) {
```

Comment:
This change will incorrectly warn that the cluster has reporting indices managed by a custom policy for every "traditional" type of deployment. 

In `Stack Management > Upgrade Assistant > Review deprecated settings and resolve issues`:

1. <img width="1338" alt="image" src="https://github.com/elastic/kibana/assets/908371/8a4e2aa8-7ee4-4588-87b4-bb71da3fd29d">
2. <img width="1338" alt="image" src="https://github.com/elastic/kibana/assets/908371/9becf5d3-a4c8-484d-8d4d-373beb397f94">
3. <img width="1338" alt="image" src="https://github.com/elastic/kibana/assets/908371/f6b576b3-1ac0-4326-b826-83e28d9a4ef2">

Reply:
Ah woops misinterpreted that! Removed it from the conditional. thanks!

Reply:
Also reverted the changes I had made on the snapshots in the test file - it should. match with the main branch now